internal/requests/requests.go

```diff
@@ -0,0 +1,254 @@
+package requests
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"mime/multipart"
+	"net/http"
+	"net/url"
+	"strings"
+
+	webscan "github.com/Method-Security/webscan/generated/go"
+)
+
+func PerformRequestScan(baseURL, path, method string, params webscan.RequestParams, vulnTypes []string) webscan.RequestReport {
```

Comment:
I assume you are only doing GET stuff for now since you are only dealing with URL Params?